<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageTaggerWidget</class>
 <widget class="QWidget" name="ImageTaggerWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>355</width>
    <height>1116</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image Tagger</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QSplitter" name="splitterMain">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="sizes" stdset="0">
      <number>667</number>
     </property>
     <widget class="QWidget" name="taggingAreaWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayoutTaggingArea">
       <item>
        <layout class="QGridLayout" name="gridLayoutApiOptions">
         <item row="0" column="0">
          <widget class="QLabel" name="labelAPI">
           <property name="text">
            <string>API:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comboBoxAPI"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelModel">
           <property name="text">
            <string>モデル:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="comboBoxModel"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelTagFormat">
           <property name="text">
            <string>FORMT:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="comboBoxTagFormat"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="lowRescheckBox">
         <property name="text">
          <string>API負荷軽減用低解像度画像を使用</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxPrompt">
         <property name="title">
          <string>プロンプト</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QTextEdit" name="textEditMainPrompt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>2</verstretch>
             </sizepolicy>
            </property>
            <property name="placeholderText">
             <string>プロンプトを入力 (例: 高画質, 具体的な描写など)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="textEditAddPrompt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>1</verstretch>
             </sizepolicy>
            </property>
            <property name="placeholderText">
             <string>選択した画像の大まかな傾向を指示するプロンプト</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonGenerate">
            <property name="text">
             <string>タグ/キャプション生成</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxResults">
         <property name="title">
          <string>結果</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="4" column="0">
           <widget class="QSlider" name="scoreSlider">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;スコア&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QTextEdit" name="textEditCaption"/>
          </item>
          <item row="1" column="0">
           <widget class="QTextEdit" name="textEditTags"/>
          </item>
          <item row="6" column="0">
           <layout class="QHBoxLayout" name="horizontalLayoutSave">
            <item>
             <widget class="DirectoryPickerWidget" name="DirectoryPickerSave" native="true"/>
            </item>
            <item>
             <widget class="QPushButton" name="pushButtonSave">
              <property name="text">
               <string>保存</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <widget class="QWidget" name="savecheckWidget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QCheckBox" name="checkBoxText">
               <property name="text">
                <string>txt</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="checkBoxJson">
               <property name="text">
                <string>Json</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="checkBoxDB">
               <property name="text">
                <string>DataBase</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QTextEdit" name="textEditGenaiPrompt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>1</verstretch>
             </sizepolicy>
            </property>
            <property name="placeholderText">
             <string>生成用Prompt｡ &quot;, &quot;で分割されてタグとして登録される</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="imageAreaWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayoutImageArea">
       <item>
        <widget class="TagFilterWidget" name="dbSearchWidget" native="true"/>
       </item>
       <item>
        <widget class="QSplitter" name="splitterImage">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizes" stdset="0">
          <number>100</number>
         </property>
         <widget class="ImagePreviewWidget" name="ImagePreview" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>1</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
         </widget>
         <widget class="ThumbnailSelectorWidget" name="ThumbnailSelector" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>2</horstretch>
            <verstretch>2</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImagePreviewWidget</class>
   <extends>QWidget</extends>
   <header location="global">ImagePreviewWidget</header>
   <container>1</container>
   <slots>
    <slot>load_image()</slot>
    <slot>update_model_options()</slot>
    <slot>send_vision_prompt()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>ThumbnailSelectorWidget</class>
   <extends>QWidget</extends>
   <header location="global">ThumbnailSelectorWidget</header>
   <container>1</container>
   <slots>
    <signal>on_thumbnail_clicked()</signal>
    <signal>imageSelected()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>DirectoryPickerWidget</class>
   <extends>QWidget</extends>
   <header location="global">DirectoryPickerWidget</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TagFilterWidget</class>
   <extends>QWidget</extends>
   <header location="global">TagFilterWidget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
